crates/ef-testing/src/sequencer/setup.rs

```diff
@@ -0,0 +1,263 @@
+use blockifier::abi::abi_utils::{
+    get_erc20_balance_var_addresses, get_uint256_storage_var_addresses,
+};
+use blockifier::state::state_api::{State, StateResult};
+use reth_primitives::{Address, Bytes};
+use revm_primitives::U256;
+use starknet_api::core::Nonce;
+use starknet_api::hash::StarkFelt;
+use starknet_api::StarknetApiError;
+
+use super::constants::{
+    CONTRACT_ACCOUNT_CLASS_HASH, EOA_CLASS_HASH, FEE_TOKEN_ADDRESS, KAKAROT_ADDRESS,
+    PROXY_CLASS_HASH,
+};
+use super::types::FeltSequencer;
+use super::utils::{
+    class_hash_to_starkfelt, contract_address_to_starkfelt, get_storage_var_address,
+    split_bytecode_to_starkfelt, split_u256,
+};
+use super::KakarotSequencer;
+
+pub trait InitializeEvmState {
+    fn initialize_contract(
+        &mut self,
+        evm_address: &Address,
+        bytecode: &Bytes,
+        nonce: U256,
+        storage: Vec<(U256, U256)>,
+    ) -> StateResult<()>;
+
+    fn fund(&mut self, evm_address: &Address, balance: U256) -> StateResult<()>;
+}
+
+impl InitializeEvmState for KakarotSequencer {
+    fn initialize_contract(
+        &mut self,
+        evm_address: &Address,
+        bytecode: &Bytes,
+        nonce: U256,
+        evm_storage: Vec<(U256, U256)>,
+    ) -> StateResult<()> {
```

Comment:
why pushing heaps of times and not only once?

Reply:
To make the code more readable

Reply:
just like inlining a loop is more readable?

Reply:
not sure what you mean, but you want me to change this then?